test/e2e/plain_provisioner_test.go

```diff
@@ -351,115 +351,100 @@ var _ = Describe("plain provisioner bundle", func() {
 
 	When("Bundles are backed by a git repository", func() {
 		var (
-			bundles []*rukpakv1alpha1.Bundle
-			ctx     context.Context
+			ctx context.Context
 		)
 		BeforeEach(func() {
 			ctx = context.Background()
+		})
 
-			By("creating the git based Bundles")
-			bundles = []*rukpakv1alpha1.Bundle{
-				{
-					ObjectMeta: metav1.ObjectMeta{
-						GenerateName: "combo-git-commit",
-					},
-					Spec: rukpakv1alpha1.BundleSpec{
-						ProvisionerClassName: plainProvisionerID,
-						Source: rukpakv1alpha1.BundleSource{
-							Type: rukpakv1alpha1.SourceTypeGit,
-							Git: &rukpakv1alpha1.GitSource{
-								Repository: "https://github.com/exdx/combo-bundle",
-								Ref: rukpakv1alpha1.GitRef{
-									Commit: "9e3ab7f1a36302ef512294d5c9f2e9b9566b811e",
-								},
-							},
+		DescribeTable("Can create and unpack a git bundle successfully",
+			func(bundle *rukpakv1alpha1.Bundle) {
+				err := c.Create(ctx, bundle)
+				Expect(err).To(BeNil())
+
+				Eventually(func() bool {
+					if err := c.Get(ctx, client.ObjectKeyFromObject(bundle), bundle); err != nil {
+						return false
+					}
+					return bundle.Status.Phase == rukpakv1alpha1.PhaseUnpacked
+				}).Should(BeTrue())
+
+				Eventually(func() bool {
+					if err := c.Get(ctx, client.ObjectKeyFromObject(bundle), bundle); err != nil {
```

Comment:
Is there a reason we're retrieving the `bundle` twice? We did it in the eventually above this ^ It could be possible to refactor this to only perform a `Get` once unless you've already thought of that.

Reply:
Yup, that makes sense. We can as well wrap everything in one Eventually().